app/anti_virus_check.py

```diff
@@ -0,0 +1,136 @@
+                time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
+            elif not results.safe:
+                error = "Unsafe file detected for case id {} with filename {}".format(payload.case_id,
+                                                                                      payload.file_name)
+                self._write_scan_report(results, payload.file_name)
+                self.bound_logger.error(error)
+                raise QuarantinableError()
+            else:
+                self.bound_logger.debug(
+                    "File {} for case id {} has been virus checked and confirmed safe".format(payload.case_id,
+                                                                                              payload.file_name))
+                return True
+
+        # out of attempts raise retryable error to force the response back to the queue.
+        self.bound_logger.error(
+            "Unable to get results of Anti-virus scan for case id {} and file {} ".format(payload.case_id,
+                                                                                          payload.file_name))
+        raise RetryableError()
+
+    def _add_api_key(self, headers):
+        # the API key is only needed for the online cloud service
+        if settings.ANTI_VIRUS_API_KEY:
+            self.bound_logger.debug("Setting A/V API key")
+            headers['apikey'] = settings.ANTI_VIRUS_API_KEY
+
+    def _check_av_response(self, response):
+        if response.status_code != 200:
+            if response.status_code == 401:
+                self.bound_logger.critical("Invalid OPSWAT API Key - unable to continue")
+                raise RetryableError()
+            elif response.status_code == 404:
+                # this could mean that the primary A/V server has failed and we've failed over to the backup
+                # in this scenario we need to start over
+                self.bound_logger.critical("OPSWAT AV does not know about this scan - the primary server may have failed")
+                raise RetryableError()
+            elif 400 <= response.status_code < 500:
+                raise BadMessageError()
+            elif 300 < response.status_code >= 500:
+                raise RetryableError()
+        else:
+            self.bound_logger.info("Response from AV OK")
+
+    def _send_for_anti_virus_check(self, filename, contents):
+        url = settings.ANTI_VIRUS_BASE_URL + "file"
+        headers = {
+            "filename": filename,
+        }
+        self._add_api_key(headers)
+
+        self.bound_logger.info("Sending for A/V scan", url=url)
+        response = requests.post(url=url, headers=headers, data=contents)
+
+        self._check_av_response(response)
+
+        self.bound_logger.info("Response received {}".format(response.text))
+        result = response.json()
+
+        if result.get("err"):
+            self.bound_logger.error("Unable to send file for anti virus scan: {}".format(result.get("err")))
```

Comment:
Use key:value pairs so that splunk indexes these correctly.

Reply:
Done